static/js/containers/pages/CheckoutPage.js

```diff
@@ -202,16 +206,28 @@ export class CheckoutPage extends React.Component<Props, State> {
 
   render() {
     const { basket, requestPending } = this.props
-    const { errors } = this.state
+    const { errors, isLoading } = this.state
 
     const item = basket && basket.items[0]
     if (!basket || !item) {
       return (
         <DocumentTitle title={`${SETTINGS.site_name} | ${CHECKOUT_PAGE_TITLE}`}>
-          <div className="checkout-page">
-            No item in basket
-            {formatErrors(errors)}
-          </div>
+          {!isLoading ? (
+            <div className="checkout-page">
+              No item in basket
+              {formatErrors(errors)}
+            </div>
+          ) : (
+            <div className="checkout-page  checkout-loader text-center align-self-center">
```

Comment:
Do we need to add / update a JS test for that HTML block rendering ?

Reply:
yes, I have added a unit test for it in recent commit.